eth/sync.go

```diff
@@ -149,73 +153,146 @@ func (pm *ProtocolManager) txsyncLoop64() {
 	}
 }
 
-// syncer is responsible for periodically synchronising with the network, both
-// downloading hashes and blocks as well as handling the announcement handler.
-func (pm *ProtocolManager) syncer() {
-	// Start and ensure cleanup of sync mechanisms
-	pm.blockFetcher.Start()
-	pm.txFetcher.Start()
-	defer pm.blockFetcher.Stop()
-	defer pm.txFetcher.Stop()
-	defer pm.downloader.Terminate()
+// chainSyncer coordinates blockchain sync components.
+type chainSyncer struct {
+	pm          *ProtocolManager
+	force       *time.Timer
+	forced      bool // true when force timer fired
+	peerEventCh chan struct{}
```

Comment:
This `peerEventCh` seems to just be a boring notification channel that something happened and we should recheck. There's no information content in it, and if there are 3 changes simultaneously, there's no need to handle each separately.

My suggestion would be to turn this into a buffered notification channel on 1 and whenever sending into it, if it's already notified, just return early. Wouldn't that make more sense?